### ISSUE TITLE
FTP grabs non-functional WARCs

### ISSUE THREAD

**Issue by rwoodpecker** [chfoo/wpull#317]
I'm passing this along from ludios/grab-site/issues/87

Here's the text:

> First off, apologies if this is intended behaviour (or inherent in WARCs or something but it doesn't seem right to me)
> 
> I've just tried a few grabs tonight on FTPs. Latest version of grab-site, no special arguments. Everything finishes as expected. However the WARCs just seem to be broken. If I try to extract the files with warcat or an unzip tool that handles WARCs.. nothing comes out. I also tried a warcat verify and it reports thousands of 'problems', one for every record.
> 
> File "/usr/local/lib/python3.4/dist-packages/warcat/tool.py", line 345, in verify_concurrent_to
> record_id), major=False)
> warcat.tool.VerifyProblem: ('Concurrent Record ID urn:uuid: not seen yet', None, False)
> 
> webarchiveplayer literally displays the WARC as empty - but the file size is proportional to the grab.
> 
> I can reproduce this across any FTP on any server. HTTP(S) grabs have no such problems and everything works as expected, verifies and extracts just fine.

**Comment by chfoo** [comment]
It looks like the WARC records are not written in the correct order or the `WARC-Concurrent-To` field needs to be placed on the latter WARC record. It's a practical problem but the resulting WARC file should still conform to the WARC spec (Section 5.7).

Does Webarchiveplayer support FTP? This issue shouldn't prevent displaying the actual file.

**Comment by rwoodpecker** [comment]
> Does Webarchiveplayer support FTP? This issue shouldn't prevent displaying the actual file.

Not sure - possibly not. Should I expect warcat to be to extract the files as well?

**Comment by chfoo** [comment]
No, Warcat only expects HTTP records. It shouldn't take too long to support them but I haven't been focusing on working on it.
